python/ray/experimental/serve/queues.py

```diff
@@ -220,3 +229,167 @@ def flush(self):
+        self.round_robin_iter[service] = itertools.cycle(backend_names)
+        self.flush()
+
+    def _flush_service_queue(self):
+        # perform traffic splitting for requests
+        for service, queue in self.queues.items():
+            # if there are incoming requests and there are backends
+            if len(queue) and len(self.traffic[service]):
+                if service not in self.round_robin_iter:
+                    backend_names = list(self.traffic[service].keys())
+                    self.round_robin_iter[service] = itertools.cycle(
+                        backend_names)
+
+                while len(queue):
+                    # choose the next backend available from persistent
+                    # information
+                    chosen_backend = next(self.round_robin_iter[service])
+                    request = queue.popleft()
+                    self.buffer_queues[chosen_backend].add(request)
+
+
+@ray.remote
+class RoundRobinPolicyQueueActor(RoundRobinPolicyQueue,
+                                 CentralizedQueuesActor):
+    pass
+
+
+class PowerOfTwoPolicyQueue(CentralizedQueues):
+    """
+    A wrapper class for powerOfTwo policy.
+    This backend selection policy is `Stateless` meaning
+    the current decisions of selecting backend are
+    dependent on previous decisions.
+    """
+
+    def _flush_service_queue(self):
+        # perform traffic splitting for requests
+        for service, queue in self.queues.items():
+            # while there are incoming requests and there are backends
+            while len(queue) and len(self.traffic[service]):
+                backend_names = list(self.traffic[service].keys())
+                backend_weights = list(self.traffic[service].values())
+                if len(self.traffic[service]) >= 2:
+                    # randomly pick 2 backends
+                    backend1, backend2 = np.random.choice(
+                        backend_names, 2, p=backend_weights)
+                    backend1, backend2 = backend1.squeeze(), backend2.squeeze()
+                    # see the length of buffer queues of the two backends
+                    # and pick the one which has less no. of queries
+                    # in the buffer
+                    if (len(self.buffer_queues[backend1]) <= len(
+                            self.buffer_queues[backend2])):
+                        chosen_backend = backend1
+                    else:
+                        chosen_backend = backend2
+                else:
+                    chosen_backend = np.random.choice(
+                        backend_names, p=backend_weights).squeeze()
+                request = queue.popleft()
+                self.buffer_queues[chosen_backend].add(request)
+
+
+@ray.remote
+class PowerOfTwoPolicyQueueActor(PowerOfTwoPolicyQueue,
+                                 CentralizedQueuesActor):
+    pass
+
+
+class FixedPackingPolicyQueue(CentralizedQueues):
```

Comment:
Can you explain a bit with examples

Reply:
You want example of FixedPackingPolicy?

Reply:
I mean this need more extended doctoring. Discussing the packing strategy a bit more. I think it's also time to add documentation in `doc/source/serve.rst` using sphinx `.. autoclass::` directives https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html